docs/articles/documentation/test-api/selecting-page-elements/selectors.md

```diff
@@ -138,45 +141,162 @@ You can initialize a selector with any of these objects.
+Property | Type | Description
+------ | ----- | -----
+`exists` | Boolean | `true` if at least one matching element exists.
+`count` | Number | The number of matching elements.
+
+```js
+import { Selector } from 'testcafe';
+import { expect } from 'chai';
 
-The [text](selector-options.md#optionstext) option specifies text content of the node that should be returned
-or a regular expression that matches this content.
+fixture `Example page`
+    .page `http://devexpress.github.io/testcafe/example/`;
 
-The [index](selector-options.md#optionsindex) option specifies the index of the node that should be returned.
+test('My test', async t => {
+    const osCount            = await Selector('.column.col-2 label').count;
+    const submitButtonExists = await Selector('#submit-button').exists;
 
-The following example demonstrates how to use the `text` and `index` options.
+    expect(osCount).eql(3);
+    expect(submitButtonExists).is.true;
+});
+
+```
+
+Note that selector property getters are asynchronous.
+
+## Combined Selectors
+
+The selector API provides methods that can be combined together, thus
+providing you with a flexible functional-style selector mechanism.
+
+### Filter Multiple DOM Nodes
+
+If selector returns multiple DOM nodes, you must filter them to select
+a single node that will eventually be returned by the selector.
+The selector provides methods to filter DOM nodes by their index or text.
+
+Method | Type | Description
+------ | ----- | -----
+`nth(index)` | Selector | Creates a selector that returns an element by its index in the matching set.
+`withText(text)` | Selector | Creates a selector that filters a matching set by the specified text.
+`withText(re)` | Selector | Creates a selector that filters a matching set using the specified regular expression.
 
 ```js
 import { expect } from 'chai';
 import { Selector } from 'testcafe';
 
-fixture `My fixture`
-    .page `http://www.example.com/`;
 
-const removeButton = Selector(cl => document.getElementsByClassName(cl), { text: 'Remove' });
-const secondButton = Selector(cl => document.getElementsByClassName(cl), { index: 1 });
+fixture `Example page`
+    .page `http://devexpress.github.io/testcafe/example/`;
 
-test('A shadowed remove button is focused', async t => {
-    expect(await removeButton('shadowed').focused).to.be.true;
+test('My test', async t => {
+    const secondCheckBox = Selector('input[type=checkbox]').nth(1);
+    const windowsLabel   = Selector('label').withText('Windows');
+
+    await t
+        .click(secondCheckBox)
+        .click(windowsLabel);
 });
+```
+
+If all nodes are filtered out, the selector returns `null`.
+
+### Find Elements by DOM Hierarchy
+
+The selector API provides methods to find elements within a DOM hierarchy in jQuery style.
+
+Property | Description
+------ | -----
+`find(cssSelector)` | Finds all descendants of all nodes in the matching set and filters them by `cssSelector`.
```

Comment:
It would be better to split tables by method groups `find`, `parent`, etc. Also it would be nice to have links for them in table of contents for quick navigation